contrib/src/main/java/com/datatorrent/contrib/cassandra/ConnectionStateManager.java

```diff
@@ -0,0 +1,471 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.cassandra;
+
+import java.io.Serializable;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hadoop.classification.InterfaceStability;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.Host;
+import com.datastax.driver.core.Metadata;
+import com.datastax.driver.core.ProtocolVersion;
+import com.datastax.driver.core.QueryOptions;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.policies.DCAwareRoundRobinPolicy;
+import com.datastax.driver.core.policies.DowngradingConsistencyRetryPolicy;
+import com.datastax.driver.core.policies.ExponentialReconnectionPolicy;
+import com.datastax.driver.core.policies.LoadBalancingPolicy;
+import com.datastax.driver.core.policies.ReconnectionPolicy;
+import com.datastax.driver.core.policies.RetryPolicy;
+import com.datastax.driver.core.policies.TokenAwarePolicy;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * This is used to specify the connection parameters that is used to process mutations for the
+ * {@link AbstractUpsertOutputOperator}.
+ * The Connection can only be set for one table in a given space and as such is used to define the following
+ * - Connection Parameters
+ * - Defaults to be used in case the execution context tuple {@link UpsertExecutionContext} does not want to
+ * set one explicitly.
+ * Note that the {@link ConnectionBuilder} is used to build an instance of the ConnectionStateManager.
+ * An instance of this class is typically instantiated by implementing the following inside
+ * {@link AbstractUpsertOutputOperator} withConnectionBuilder() method.
+ * {@code
+      @Override
+      public ConnectionStateManager.ConnectionBuilder withConnectionBuilder()
+      {
+        return ConnectionStateManager.withNewBuilder()
```

Comment:
I have added functionality to read from a properties file. However here are my thoughts on this: 

There are pros and cons with a properties file approach.

Pros : 
- Easier to "configure" by writing the properties file. However the concrete implementation of the operator still needs to implement abstract methods and hence as well might need some code to be in place. Hence the gain is not that much. 

Cons:
- Severely restricts that there is only one operator in the container that can output to cassandra. Since this operator is severely tied down to a single table and keyspace at instantiation time and these properties are getting loaded from a standard file name, we will be restricting that only one logical Cassandra Upsert operator can be used  at any given point in time. 

However the implementation allows the end user to use the builder pattern to override the properties file based approach.